### PR TITLE
Content for 'search for something'

### PR DESCRIPTION
**Pull request by devansXD**
Content from initial draft. Needs approval.

### CODE REVIEW

**Comment by daniel-ac-martin** [comment]
@MarionDownie: Are you happy to review https://61e70b31acb7ff0cd3b77ed5--hods.netlify.app/patterns/search-for-something ?

File: apps/docs/src/common/pages/patterns/stop-a-service-timing-out.tsx [hunk 1]
```diff
@@ -47,6 +47,10 @@ const Page: FC<PageProps> = ({ location }) => (
         {
           href: '/patterns/stop-a-service-timing-out',
           text: 'Stop a service timing out'
+        },
+        {
+          href: '/patterns/search-for-pattern',
```
Comment (daniel-ac-martin) [inline apps/docs/src/common/pages/patterns/stop-a-service-timing-out.tsx:52]: pattern => something

File: apps/docs/src/common/pages/patterns/search-for-something.tsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+      <h2 className="govuk-heading-l">Accessibility considerations</h2>
+      <p>Ensure you test your search with users with access needs. </p>
+
+      <h3 className="govuk-heading-m">Keyboard navigation</h3>
+
+      <p>Tabbing in a logical way to the interactive elements, being able to see the focus, and being able to interact with it. If not using native HTML elements and creating something custom refer to <a href="https://www.w3.org/TR/wai-aria-practices/wai-aria-practices">https://www.w3.org/TR/wai-aria-practices/wai-aria-practices</a> for expected interaction., </p>
+
+      <h3 className="govuk-heading-l">Screen readers</h3>
+      <ul className="govuk-list govuk-list--bullet"><li>Ensure programmatic labelling of fields</li>
+        <li>Ensure that all interactive elements have a name, role and value, see WAI-ARIA Authoring Practices 1.1 for expected values.</li>
```
Comment (guermanc) [inline apps/docs/src/common/pages/patterns/search-for-something.tsx:144]:
```suggestion
      <ul className="govuk-list govuk-list--bullet"><li>ensure programmatic labelling of fields</li>
        <li>ensure that all interactive elements have a name, role and value, see WAI-ARIA Authoring Practices 1.1 for expected values</li>
```

File: apps/docs/src/common/pages/patterns/search-for-something.tsx [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+
+      <h3 className="govuk-heading-l">Other</h3>
+      <ul className="govuk-list govuk-list--bullet"><li>when presenting data in tables, ensure this is marked up correctly</li>
+        <li>make sure colour only isn't used to assign meaning to results</li>
+        <li>make sure you can resize text up to 200% and scale up to 400% (tables excluded) respectively. </li>
+        <li>ensure that users can change font spacing. All of these especially important if there is a lot of data.</li>
+        <li>allow users to turn off, adjust or extend timeouts</li>
+        <li>especially for filtering, make sure not to change results based on filter checkbox selection alone for example, provide a submit button</li>
+        <li>If the search generates errors, making sure screen reader users know what happened and how to fix it.</li>
```
Comment (guermanc) [inline apps/docs/src/common/pages/patterns/search-for-something.tsx:155]: Would rephrase colour sections as something like "make sure colour is not used as the only way to assign meaning to results".

Would just mention resizing content to 400% to simplify requirement and avoid ambiguity.

Remove initial capital letter and full stops at the end of list items to be more in line with other lists. Change "making" in the last list item to "make".

**Review by guermanc — COMMENTED** [review]
Added comments on punctuation errors, some leftover references to the stop a service from timing out page and rewording suggestions for some accessibility points.

File: apps/docs/src/common/pages/patterns/stop-a-service-timing-out.tsx (hunk 1 above)
Comment (daniel-ac-martin) [reply apps/docs/src/common/pages/patterns/stop-a-service-timing-out.tsx:52]:
```suggestion
          href: '/patterns/search-for-something',
```
Just discovered this was a thing! (Thanks @guermanc.)

**Comment by MarionDownie** [comment]
Emailing across changes

**Comment by devansXD** [comment]
All comments now resolved.

**Comment by MarionDownie** [comment]
You left my comments in Dave! Sorry if wasn't clear. Should read as follows (without bold anywhere):

If your users normally have a single key piece of information to hand to search with, such as a reference number or identifier, allow them to search by that as it will allow for a quick match. When an exact match is found, surface the match result on the screen (rather than going directing to the record).

File: apps/docs/src/common/pages/patterns/search-for-something.tsx [hunk 4]
```diff
@@ -0,0 +1,176 @@
+import { FC, createElement as h } from 'react';
+import { Helmet } from 'react-helmet-async';
+import { PageProps } from '@not-govuk/app-composer';
+import { A, NavigationMenu } from '@not-govuk/components';
+
+ const applySearchNarrow = require('../../../../assets/images/patterns/search-narrow.svg').default;
+ const applySearchWide= require('../../../../assets/images/patterns/search-wide.svg').default;
+
+export const title = 'Search for something';
+const description = 'How to help users to stop a service from timing out';
```
[the remark this answers is not in the document]
Comment (daniel-ac-martin) [reply apps/docs/src/common/pages/patterns/search-for-something.tsx:10]: Should this have been resolved? - It looks like something that should be fixed to me.

[the remark this answers is not in the document]
Comment (daniel-ac-martin) [reply apps/docs/src/common/pages/patterns/search-for-something.tsx:10]: @devansXD: What do you think?

[the remark this answers is not in the document]
Comment (devansXD) [reply apps/docs/src/common/pages/patterns/search-for-something.tsx:10]: Yes, not sure why not showing. Should be fine now.

[the remark this answers is not in the document]
Comment (devansXD) [reply apps/docs/src/common/pages/patterns/search-for-something.tsx:10]: @daniel-ac-martin

**Comment by devansXD** [comment]
@daniel-ac-martin @MarionDownie fixed the missing navigation panel item, on search pattern and stop a service timing out.

**Comment by MarionDownie** [comment]
Still not working for me - Daniel to check.

Also 'stop a service timing out' seems to think to the wrong thing? Link broken.

**Comment by daniel-ac-martin** [comment]
I've opened a new PR for any fixes: #326
